tsnet/tsnet.go

```diff
@@ -94,17 +95,25 @@ func (s *Server) Start() error {
 	return s.initErr
 }
 
+// Close stops the server.
+//
+// It must not be called before or concurrently with Start.
+func (s *Server) Close() error {
+	s.shutdownCancel()
+	s.lb.Shutdown()
+	s.linkMon.Close()
+	return nil
+}
```

Comment:
we should also close the `nettest.Listener`
It also overrides the global `tailscale.TailscaledDialer`, so maybe set that to the old value? I really want `tailscale.Client`, so callers can decide what to do

Reply:
Done. 

About the dialer: oh well. Can clean later.